test/testdata/infer/private_result_type_as_seen_from.rb

```diff
@@ -0,0 +1,28 @@
+# typed: true
+
+class Parent
+  extend T::Sig
+  extend T::Generic
+  Elem = type_member
+
+  sig {returns(Elem)}
+  attr_accessor :elem
+
+  sig {returns(T.self_type)}
+  def get_self; self; end
+end
+
+class Child < Parent
+  Elem = type_member(fixed: Integer)
+
+  private :elem
+  private :get_self
+
+  def example
+    T.reveal_type(elem) # error: Revealed type: `Integer`
+
+    self.elem = '' # error: Assigning a value to `elem` that does not match expected type `Integer`
+
+    T.reveal_type(get_self) # error: Revealed type: `Child`
+  end
+end
```

Comment:
My first approach was to register an intrinsic that behaved somewhat like `<Magic>.<self-new>` and simulated the dispatch call on the super method.

I abandoned that approach halfway through when I realized that `dispatchCallSymbol` takes a `targs` vector but we don't have a targs vector in the `apply` callback of IntrinsicMethod, and even if we did, I didn't want to have to futz around with aligning those type args as if they were seen from the parent class.